### ISSUE TITLE
no-archive model cannot be loaded

### ISSUE THREAD

**Issue by vikigenius** [pytorch/serve#721]
## Context

* torchserve version: 0.2.0
* torch version: 1.6.0
* java version: 11.0.9
* Operating System and version: GNU/Linux

## Your Environment

* Installed using source? [no]:
* Are you planning to deploy it using docker container? [no]:
* Is it a CPU or GPU environment?: GPU
* Using a default/custom handler? [custom]:
* What kind of model is it e.g. vision, text, audio?: text
* Are you planning to use local models from model-store or public url being used e.g. from S3 bucket etc.?
  [If public url then provide link.]: local

## Expected Behavior

No archive model should be able to be loaded

## Current Behavior

Model not found error when loading a no-archive model

## Steps to Reproduce

1. Create a no-archive version of the model using model-archiver

    `torch-model-archiver --model-name "BartQgen" --version 1.0 --serialized-file models/sqgen/pytorch_model.bin --handler sqgen/handler.py --extra-files "models/sqgen/config.json,models/sqgen/special_tokens_map.json,models/sqgen/tokenizer_config.json,models/sqgen/vocab.json,models/sqgen/merges.txt" --archive-format no-archive --export-path model_store`

2. Try to load it using torchserve:
    `torchserve --start --model-store model_store --models sqgen=BartQgen --no-config-snapshots `

## Failure Logs [if any]

```
2020-10-09 20:12:08,346 [INFO ] main org.pytorch.serve.ModelServer - Loading initial models: BartQgen
2020-10-09 20:12:08,348 [WARN ] main org.pytorch.serve.ModelServer - Failed to load model: BartQgen
org.pytorch.serve.archive.ModelNotFoundException: Model not found at: BartQgen
        at org.pytorch.serve.archive.ModelArchive.downloadModel(ModelArchive.java:84)
        at org.pytorch.serve.wlm.ModelManager.createModelArchive(ModelManager.java:137)
        at org.pytorch.serve.wlm.ModelManager.registerModel(ModelManager.java:115)
        at org.pytorch.serve.ModelServer.initModelStore(ModelServer.java:213)
        at org.pytorch.serve.ModelServer.start(ModelServer.java:308)
        at org.pytorch.serve.ModelServer.startAndWait(ModelServer.java:104)
        at org.pytorch.serve.ModelServer.main(ModelServer.java:85)
```

I also tried providing the absolute path to the model. That doesn't work either.

**Comment by vikigenius** [comment]
@dhaniram-kshirsagar where you able to reproduce the issue?

**Comment by reubenwenisch** [comment]
Can you try and comment on what you get
`torchserve --start --model-store model_store --models sqgen=BartQgen.mar --no-config-snapshots`

**Comment by vikigenius** [comment]
@reubenwenisch if I use BartQgen.mar, I get the same model not found error. Of course it works perfectly fine if i actually create an archive called BartQgen.mar by removing the no-archive flag.

**Comment by dhaniram-kshirsagar** [comment]
@vikigenius It is a bug. We will fix it in 0.3 release.
This is the [line](https://github.com/pytorch/serve/blob/master/frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java#L87) which is causing problem. It needs to be replaced with a load call instead of an exception.

**Comment by harshbafna** [comment]
Naked DIR support has been merged as part of #862. Closing.

**Comment by Ark-kun** [comment]
The change breaks the error reporting.
It's very confusing when the system tells you that file does not exist even though it clearly does.
